examples/cloudproviders/k8s/daemonset.yaml

```diff
@@ -0,0 +1,116 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: gostatsd-cfg
+  namespace: default
+data:
+  config.toml: |
+    # Example: don't actually do anything with the metrics
+    backends=['stdout']
+    default-tags=[]
+    internal-tags=['in_cluster:true']
+    statser-type='internal'
+    flush-interval='1s'
+    bad-lines-per-minute=1000
+    metrics-addr=':8125'
+    filters='no-hostname'
+
+    [k8s]
+    # Include any annotation beginning with myapp.company.com as a tag, using the part after the / as the tag name
+    annotation-tag-regex = '^myapp.company.com/(?P<tag>.*)$'
+    # Include the 'app' label as a tag
+    label-tag-regex = '^app$'
+    # Only watch pods that are on the node that this gostatsd is running on.
+    watch-cluster = false
+    # This MUST be overridden via environment variable GSD_K8S_NODE_NAME. See pod spec for example.
+    node-name = ''
+
+    [filter.no-hostname]
+    # Drop the hostname from metrics because it can be very noisy with pod churn on Kubernetes
+    drop-host=true
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: gostatsd
+  namespace: default
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: gostatsd
+  namespace: default
+  labels:
+    app: gostatsd
+spec:
+  selector:
+    matchLabels:
+      app: gostatsd
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: gostatsd
+      annotations:
+        myapp.company.com/imageVersion: "latest"
+    spec:
+      serviceAccountName: gostatsd
+      # We need the host network so we have an explicitly addressable pod from every pod running on the node.
```

Comment:
Something I hadn't considered - one of the big problems with statsd is that almost all clients do "resolve one, refresh never", which is a massive pain.  What are the implications of updating the DaemonSet?  Will the IP remain constant because of `hostNetwork` being true?

Reply:
Yes, the IP remains constant. The old pod is terminated before the new one can be started because both pods cannot share the same IP:port combination.